pxfish/protocol_test.py

```diff
@@ -1,16 +1,22 @@
 """
 code for running operation type tests
 """
+import json
 import logging
+import os
 
-
-def parse_test_response(response):
+def parse_test_response(*, response, file_path):
     """
     Reports test results
+    Writes errors to screen and creates json file with backtrace
 
     Arguments:
         response (dict): response received after running test
     """
+    file_path = os.path.join(file_path, 'test_results.json')
```

Comment:
How about putting the file dump into a different function, since this method is called `parse_test_response`